pkg/orchestration/wiring/postgres/plugin.go

```diff
@@ -72,22 +72,32 @@ func New() *WiringPlugin {
 	}
 }
 
+// getRDSDependency returns a single RDS dependency if found in a given list, and will return an error if multiple are found
 func getRDSDependency(dependencies []wiringplugin.WiredDependency) (wiringplugin.WiredDependency, error) {
 	rdsDependency := []wiringplugin.WiredDependency{}
-	if len(dependencies) > 0 {
-		for _, d := range dependencies {
-			if d.Type == rds.ResourceType {
-				rdsDependency = append(rdsDependency, d)
-			}
-		}
-		if len(rdsDependency) > 1 {
-			return wiringplugin.WiredDependency{}, errors.Errorf("Postgres resources can only depend on one RDS resource type")
+
+	// Collect all RDS dependencies
+	for _, d := range dependencies {
+		_, found, err := knownshapes.FindRDSShapes(d.Contract.Shapes)
+		if err != nil {
+			return wiringplugin.WiredDependency{}, errors.Wrap(err, "unable to get RDS dependency")
 		}
-		if len(rdsDependency) == 1 {
-			return rdsDependency[0], nil
+		if found {
+			rdsDependency = append(rdsDependency, d)
 		}
 	}
-	return wiringplugin.WiredDependency{}, nil
+
+	// Did not find a RDS dependency
+	if len(rdsDependency) == 0 {
+		return wiringplugin.WiredDependency{}, nil
+	}
+
+	// Ensure postgres does not depend on more than one RDS resource
+	if len(rdsDependency) > 1 {
+		return wiringplugin.WiredDependency{}, errors.Errorf("postgres resources can only depend on one RDS resource type")
+	}
+
+	return rdsDependency[0], nil
```

Comment:
I reworked this function a little bit as I find it simple to have the happy case as the "first tab" and error/unexpected cases handled in the "second tab" aka indent path.